### DISCUSSION TITLE
Who can help me change the color here?

### DISCUSSION THREAD

**Discussion by sendudesign** [Abdelrhman-AK/WinPaletter#58] [Q&A]
I use WinPaletter changed the main title bar color, I am very satisfied.
Do you have any way to change the color of the arrow is pointing to?
If very trouble, I can accept the charges
<img width="602" alt="Snipaste_2022-12-10_18-57-48" src="https://user-images.githubusercontent.com/118046810/206851751-016f1402-8ec0-45b3-82bc-bb550149bc2d.png">
<img width="591" alt="Snipaste_2022-12-10_18-57-36" src="https://user-images.githubusercontent.com/118046810/206851752-cfc8f161-fef4-4ca8-be03-74463efe3421.png">

**Comment by Abdelrhman-AK** [comment]
### It is supposed to be in Win32UI > Selection Objects > Menu (& Menu Bar)

But actually this didn't change the color, it changed a line color only as in the screenshot (In both Windows 11/10) 
![menu](https://user-images.githubusercontent.com/59510211/206852515-d253d89f-4e4e-44cd-88b8-0a2b468e9717.png)

- Win32UI's Form is oriented to change the classic items of Windows (Not all items).
- So, the menus are mainly handled by Visual Styles (Not registry colors), if you want to change this color you can search for a custom visual style online.

**Reply by sendudesign** [reply]
Thank you very much for your reply. I tried to search, but unfortunately I couldn&#39;t find the relevant information.

**Reply by sendudesign** [reply]
Where can I get Win32Ul? I want to give it a try

**Reply by Abdelrhman-AK** [reply]
In the main form, you will find that button below

![Screenshot 2022-12-10 135325](https://user-images.githubusercontent.com/59510211/206853802-f21e4183-3b2d-498c-b943-75a50dad3e07.png)

**Reply by sendudesign** [reply]
Thank you, it really doesn't achieve the effect I wanted

**Reply by Abdelrhman-AK** [reply]
You're Welcome.
If I find a solution, I will write it here.

**Reply by sendudesign** [reply]
ok

**Reply by Abdelrhman-AK** [reply]
There is a solution, but it might not be simple:
You will need 3 programs, or 4

1. [msstyleEditor - Github](https://github.com/nptr/msstyleEditor); copy `C:\Windows\Resources\Themes\aero` to a new folder and work on it.
2. PNG Editing Tools (Paint.NET, or GIMP, or Photoshop, ....)
3. [SecureUxTheme - Github](https://github.com/namazso/SecureUxTheme); and this is to apply the theme you made.
4. `Optional` A Virtual Machine to test the theme before applying to avoid crashes\bugs\undesired action on your main device.


Now, the steps:
1. Copy `C:\Windows\Resources\Themes\Aero` Folder into a new folder.
2. Use msstyleEditor to edit the visual styles, you will find the entry of menus like here
![aero.msstyles (Win 8.1) - Windows Style Builder.png](https://user-images.githubusercontent.com/59510211/206856073-1a31aaff-f8f0-496e-a786-430222531a69.png)

[Take a look about more details here](https://www.tenforums.com/customization/43571-change-color-menubar-list-header.html)

3. After exporting this png, edit it as you like, then reimport the modified png
4. Save your visual styles and test it in a virtual machine (To avoid crashes or bugs - if you don't want to use it you can skip it)
5. Use SecureUXTheme to apply the visual styles you made

**Reply by sendudesign** [reply]
You are so good. I'll try it now

**Reply by Abdelrhman-AK** [reply]
Thanks!
Have a good luck

**Reply by sendudesign** [reply]
> Thanks! Have a good luck

When I am on this step, I don't know how to lay their hands on.
I just want to change my hair the color of the pictures of the arrow
<img width="665" alt="Snipaste_2022-12-10_21-34-45" src="https://user-images.githubusercontent.com/118046810/206858037-3cbf6f92-ac84-47c0-afd7-e24574561e42.png">

**Reply by Abdelrhman-AK** [reply]
Click on "export" button above
Then edit this png with any image editor like Paint.NET or Photoshop

Then import the edited png

**Reply by sendudesign** [reply]
> 点击上面的“导出”按钮，然后使用任何图像编辑器（如 Paint.NET 或Photoshop）编辑此png
> 
> 然后导入编辑后的 png

This is really too difficult for me without a software foundation. Can you make a topic file for me? I will pay you a certain reward as a reward.
You just need to help me change the default theme and the color of the arrow (RGB 68,68,68)
<img width="84" alt="Snipaste_2022-12-10_21-44-25" src="https://user-images.githubusercontent.com/118046810/206858406-a86b9ad0-784b-4470-9727-9fa292095a17.png">

**Reply by Abdelrhman-AK** [reply]
Ok. I'll try now on my device.
What is the certain color of the bar ?
And the color of selection (instead of blue) ?

**Reply by sendudesign** [reply]
> Ok. I'll try now on my device. What is the certain color of the bar ? And the color of selection (instead of blue) ?

Their colors are all RGB:68,68,68
<img width="591" alt="Snipaste_2022-12-10_22-05-30" src="https://user-images.githubusercontent.com/118046810/206859296-518c49a8-44ed-461b-9f19-56a838f07d19.png">

**Reply by Abdelrhman-AK** [reply]
![0000](https://user-images.githubusercontent.com/59510211/206860034-823126bc-d840-4fb7-aa0c-ec455bafee29.png)
![Untitled](https://user-images.githubusercontent.com/59510211/206860036-d8bafcb9-29f1-4917-929f-8e1733b53403.png)

Done!
The item to be modified was wrong. The right one is in the screenshot

[Aero (Modified).zip](https://github.com/Abdelrhman-AK/WinPaletter/files/10200263/Aero.Modified.zip)

- Extract this ZIP to `C:\Windows\Resources\Themes\`

- Open SecureUXTheme
- Click on install and click yes (To reboot)
- Reopen SecureUXTheme
- Click on `Aero (Modified)` and press apply

**Reply by sendudesign** [reply]
Strange, I there is no change, where is my error?
<img width="1124" alt="Snipaste_2022-12-10_22-42-32" src="https://user-images.githubusercontent.com/118046810/206860905-ffc2dff5-79cd-4987-8b14-691e3ca7c4d1.png">

**Reply by Abdelrhman-AK** [reply]
That's weird. Try to re-install, patch and apply

**Reply by sendudesign** [reply]
> 这很奇怪。尝试重新安装、修补和应用

Follow your steps again, but it still doesn't work

**Reply by Abdelrhman-AK** [reply]
- Try a reboot, return to the original aero theme and reapply the modified one.

- If not, then I suspect that the chinese .mui msstyles may be the reason of this. If so, try to patch the visual styles yourself and I will upload the modified PNG I made.

**Reply by sendudesign** [reply]
When I click the test button in the software, it works, but using themes doesn't work
<img width="1301" alt="Snipaste_2022-12-10_22-58-37" src="https://user-images.githubusercontent.com/118046810/206861573-9136af4b-756e-4782-951f-2d845b618494.png">

**Reply by sendudesign** [reply]
> * 尝试重新启动，返回到原始 aero 主题并重新应用修改后的主题。
> * 如果不是，那么我怀疑中文的.mui msstyle可能是造成这种情况的原因。如果是这样，请尝试自己修补视觉样式，我将上传我制作的修改后的 PNG。

ok,thank you very much!

**Reply by Abdelrhman-AK** [reply]
Maybe SecureUXTheme didn't apply it correctly (or didn't apply it at all). Stop testing then,

- Go to `C:\Windows\Resources\Themes` and open `Aero (Modified).theme` and see if it will work or not. 

- If not, open SecureUXTheme, apply the original Aero, press on uninstall, restart your Windows, then open SecureUXTheme again and install, and restart, then open the program again and apply the modified aero.

**Reply by sendudesign** [reply]
> Maybe SecureUXTheme didn't apply it correctly (or didn't apply it at all). Stop testing then,
> 
> * Go to `C:\Windows\Resources\Themes` and open `Aero (Modified).theme` and see if it will work or not.
> * If not, open SecureUXTheme, apply the original Aero, press on uninstall, restart your Windows, then open SecureUXTheme again and install, and restart, then open the program again and apply the modified aero.

Follow your steps to operate a series, it has not changed, please send your PNG I try to modify myself

**Reply by Abdelrhman-AK** [reply]
- Here you are:
![Menu_BARBACKGROUND_1345](https://user-images.githubusercontent.com/59510211/206862261-34e79627-4fed-4e99-a28c-53fae1dc5b0a.png)

- Don't forget to modify these colors too:
![0](https://user-images.githubusercontent.com/59510211/206862269-ba0c17d5-092f-4320-8c3b-eb51b146e114.png)

**Reply by sendudesign** [reply]
Forgive my stupidity, I am a Chinese person who can not understand the interface of this software, so I would ask you how I should have to replace the black PNG picture in it

**Reply by Abdelrhman-AK** [reply]
No problem!
![Screenshot_20221210-172600_Chrome.png](https://user-images.githubusercontent.com/59510211/206862551-6352e539-89bc-417a-b1e2-d61cc20768e6.png)

**Reply by Abdelrhman-AK** [reply]
And when you save the visual styles try to press on the lower part of save, and choose the save with MUI. If it gives you errors try to save normally

![Screenshot_20221210-172851_Chrome.png](https://user-images.githubusercontent.com/59510211/206862681-a2203f89-7ca6-422e-b6d1-35034a841cff.png)

**Reply by sendudesign** [reply]
Thank you very much for your help! Now I have successfully modified its color
But now there are new problems
I set the color of the title bar in WinPaletter doesn't seem to work, lighter in third party software
<img width="1289" alt="Snipaste_2022-12-11_00-08-36" src="https://user-images.githubusercontent.com/118046810/206864337-c2ad9f45-89c3-4eda-be4d-406c59a98dc6.png">

**Reply by Abdelrhman-AK** [reply]
- That is the same issue that happened with me, EXACTLY. and I couldn't resolve it.

- This isuse maybe related to msstyleEditor itself, not WinPaletter

**Reply by sendudesign** [reply]
Maybe we can change the color of the title bar directly in msstyleEditor?

**Reply by Abdelrhman-AK** [reply]
I'm not sure if it is present or not.
Try to search for `Titlebar` in the application.

**Reply by sendudesign** [reply]
Can you continue to help me? Just one step away from perfection

**Reply by Abdelrhman-AK** [reply]
Yes, I'll help you
Did you find this entry or not ?

**Reply by sendudesign** [reply]
I can't read English, so I'm depressed

**Reply by Abdelrhman-AK** [reply]
I'll try myself and post here if I found it or not

**Reply by Abdelrhman-AK** [reply]
I tried too much. I navigated through nearly all items and couldn't find the reason for this issue. As I'm not experienced too much in editing Visual Styles, I don't know where is the responsible item.

When you set colors by WinPaletter, the active one is set, while the inactive one isn't. It gets its color from the active one (But Lighter).

![Untitled](https://user-images.githubusercontent.com/59510211/206866159-471f5531-cae3-4df9-9b35-e5ef91dccdca.png)

I tried another theme (Rectify11). Titlebars are dark but both borders (Active + Inactive) can be colorized in the same color you choose in WinPaletter, so the issue is mainly by msstyles editor itself.

[Rectify11.zip](https://github.com/Abdelrhman-AK/WinPaletter/files/10200531/Rectify11.zip)

You can contact msstylesEditor's developer if he can afford Chinese Translations or not if you're interested in designing visual styles in the future.

**Reply by sendudesign** [reply]
I see what you mean, I'm also looking for the solution, if there is progress I will inform you

**Reply by Abdelrhman-AK** [reply]
Okay!

**Reply by sendudesign** [reply]
I used their theme, as if I can't color the title bar in WinPaletter, can you?
<img width="1261" alt="Snipaste_2022-12-11_01-45-50" src="https://user-images.githubusercontent.com/118046810/206868374-8af4f930-c067-4d1a-9eb3-9f54eeb9b87c.png">

**Reply by Abdelrhman-AK** [reply]
They intentionally made it black. I meant the borders of active and intactive Windows can be colorized with the colors you choose. 

In conclusion; both colors can be used in rectify 11 but not in the one made by msstyleEditor. And that is the issue

**Reply by sendudesign** [reply]
It is very difficult to achieve the effect of perfect unity. I continue to look for other ways.

**Reply by Abdelrhman-AK** [reply]
Ok. No problem. Have good luck with this!
